extension/src/popup/components/sendPayment/SendConfirm/TransactionDetails/index.tsx

```diff
@@ -269,44 +269,19 @@ export const TransactionDetails = ({ goBack }: { goBack: () => void }) => {
 
   const handleXferTransaction = async () => {
     try {
-      const assetAddress = asset.split(":")[1];
-      const assetBalance = tokenBalances.find(
-        (b) => b.contractId === assetAddress,
-      );
-
-      if (!assetBalance) {
-        throw new Error("Asset Balance not available");
-      }
-
-      const parsedAmount = parseTokenAmount(
-        amount,
-        Number(assetBalance.decimals),
-      );
-
-      const params = [
-        new SorobanClient.Address(publicKey).toScVal(), // from
-        new SorobanClient.Address(destination).toScVal(), // to
-        new SorobanClient.XdrLargeInt("i128", parsedAmount.toNumber()).toI128(), // amount
-      ];
-
       if (!hasSorobanClient(sorobanClient)) {
         throw new Error("Soroban RPC not supported for this network");
       }
 
-      const builder = await sorobanClient.newTxBuilder(
-        xlmToStroop(transactionFee).toFixed(),
-      );
-
-      const transaction = await transfer(assetAddress, params, memo, builder);
-
-      const preparedTransaction = await sorobanClient.server.prepareTransaction(
-        transaction,
+      const preparedTransaction = SorobanClient.assembleTransaction(
```

Comment:
I think we can refactor this store so that the state is a union type where the tx stuff is no longer nullable in certain states(sim success/submit success) so that we don't have to cast these to non-nullable. I didn't think it'd be worth it in this PR but let me know if you think otherwise @piyalbasu

Reply:
Cool - yeah, I thin we can clean up the types in another PR 👍